### PR TITLE
Supports onnx LayerNormalization-17 via opset6::MVN

### PR DESCRIPTION
**Pull request by jane-intel**
### Details:
 - *LayerNormalization-17 translation got more compact*
 - *Three output translation is unavailable anymore*

### Tickets:
 - *CVS-132128*

### CODE REVIEW

File: src/frontends/onnx/frontend/src/op/layer_normalization.cpp [hunk 1]
```diff
@@ -43,58 +32,42 @@ ov::OutputVector layer_normalization(const ov::frontend::onnx::Node& node) {
...
 
-    const auto& X = inputs.at(0);
-    const auto& Scale = inputs.at(1);
-
-    auto axis = node.get_attribute_value<std::int64_t>("axis", -1);
-    double epsilon = node.get_attribute_value<double>("epsilon", 1e-5);
-    int64_t stash_type_i =
-        node.get_attribute_value<int64_t>("stash_type",
-                                          static_cast<int64_t>(TensorProto_DataType::TensorProto_DataType_FLOAT));
+    auto default_stash_type_i = static_cast<int64_t>(TensorProto_DataType::TensorProto_DataType_FLOAT);
```
Comment (slyalin) [inline src/frontends/onnx/frontend/src/op/layer_normalization.cpp:42]: Conversion code shouldn't rely on ONNX-specific data structures that should be isolated under the decoder. Is the TensorProto_DataType::TensorProto_DataType_FLOAT usage unavoidable?

File: src/frontends/onnx/frontend/src/op/layer_normalization.cpp [hunk 2]
```diff
@@ -43,58 +32,42 @@ ov::OutputVector layer_normalization(const ov::frontend::onnx::Node& node) {
...
-    auto X2D = ov::op::util::flatten(X, static_cast<int>(axis));
-    auto XU = std::make_shared<Convert>(X2D, stash_type);
-
-    auto Mean2D = std::make_shared<ReduceMean>(XU, One1D, true);
-    auto Square = std::make_shared<Multiply>(XU, XU);
-    auto MeanOfSquare = std::make_shared<ReduceMean>(Square, One1D, true);
-    auto SquareOfMean = std::make_shared<Multiply>(Mean2D, Mean2D);
+    ov::Output<ov::Node> data = inputs.at(0);
+    element::Type original_type = data.get_element_type();
+    bool needs_type_casting = original_type.is_dynamic() || stash_type != original_type;
```
Comment (slyalin) [inline src/frontends/onnx/frontend/src/op/layer_normalization.cpp:52]: Can we just compare with dynamic type without checking that it is dynamic?

File: src/frontends/onnx/frontend/src/op/layer_normalization.cpp [hunk 3]
```diff
@@ -43,58 +32,42 @@ ov::OutputVector layer_normalization(const ov::frontend::onnx::Node& node) {
...
-    int64_t stash_type_i =
-        node.get_attribute_value<int64_t>("stash_type",
-                                          static_cast<int64_t>(TensorProto_DataType::TensorProto_DataType_FLOAT));
+    auto default_stash_type_i = static_cast<int64_t>(TensorProto_DataType::TensorProto_DataType_FLOAT);
+    int64_t stash_type_i = node.get_attribute_value<int64_t>("stash_type", default_stash_type_i);
     element::Type stash_type = common::get_ov_element_type(stash_type_i);
+    CHECK_VALID_NODE(node,
+                     stash_type == element::f32 || stash_type == element::bf16,
+                     "LayerNormalization `stash_type` attribute is expected to represent float or bfloat16 type. Got: ",
+                     stash_type);
```
Comment (slyalin) [inline src/frontends/onnx/frontend/src/op/layer_normalization.cpp:48]: Why do we need this extra validation? Just to do ONNX job? We don't really care about the type, our conversion code will be the same. Next time when LayerNormalization ONNX operator's specification is update, we need to change this code on our side.

File: src/frontends/onnx/frontend/src/op/layer_normalization.cpp (hunk 1 above)
Comment (jane-intel) [reply src/frontends/onnx/frontend/src/op/layer_normalization.cpp:42]: This is default value of `stash_type` attribute. It has to do with operation semantics only.

File: src/frontends/onnx/frontend/src/op/layer_normalization.cpp (hunk 2 above)
Comment (jane-intel) [reply src/frontends/onnx/frontend/src/op/layer_normalization.cpp]: Updated code. Please review

File: src/frontends/onnx/frontend/src/op/layer_normalization.cpp (hunk 3 above)
Comment (jane-intel) [reply src/frontends/onnx/frontend/src/op/layer_normalization.cpp]: Updated code. Please review

**Review by slyalin — APPROVED** [review]

File: src/frontends/onnx/frontend/src/op/layer_normalization.cpp [hunk 4]
```diff
@@ -6,95 +6,81 @@
...
-    auto Normalized = std::make_shared<Divide>(Deviation, StdDev);
-    auto NormalizedT = std::make_shared<ConvertLike>(Normalized, X);
+    auto axis = node.get_attribute_value<std::int64_t>("axis", -1);
+    // ONNX operator semantics says that `axis` attribute points to the first normalization dimension. We have to
+    // figure out all dimensions for normalization:
+    // axis < 0:  range(axis, 0)     Example: axis = -2; Axes: [-2, -1]
+    // axis >= 0: range(axis, rank)  Example: axis = 2, rank = 4; Axes: [2, 3]
+    auto axes =
+        std::make_shared<v4::Range>(Constant::create(element::i64, {}, {axis}),
+                                    (axis < 0 ? Constant::create(element::i64, {}, {0})->output(0) : rank(data)),
```
Comment (mitruska) [inline src/frontends/onnx/frontend/src/op/layer_normalization.cpp:70]: Is there a specific reason to use `->output(0)` for the Constant here?

**Review by mitruska — APPROVED** [review]

File: src/frontends/onnx/frontend/src/op/layer_normalization.cpp (hunk 4 above)
Comment (jane-intel) [reply src/frontends/onnx/frontend/src/op/layer_normalization.cpp:70]: due to trernary operator

**Pull request merged**